src/wasm/wasm-type.cpp

```diff
@@ -471,9 +520,61 @@ bool Type::isSubType(Type left, Type right) {
   if (left == right) {
     return true;
   }
-  if (left.isRef() && right.isRef() &&
-      (right == Type::externref || left == Type::nullref)) {
-    return true;
+  if (left.isRef() && right.isRef()) {
+    // ╒═══════ HeapType ═══════╕ ╒═════ ValueType ═════╕
+    // │         extern         │ │        (any)        │
+    // │            │           │ │      ┌───┴───┐      │
+    // │           any          │ │  rtt n ht   ...     │
+    // │      ┌─────┼─────┐     │ │      │              │
+    // │    func    eq   exn    │ │  rtt n'>=n ht       │
+    // │      │     │           │ └─────────────────────┘
+    // │ signature* │           │ ╒════════ Ref ════════╕
+    // │            │           │ │     ref null ht     │
+    // │      ┌─────┼─────┐     │ │          │          │
+    // │    i31  struct* array* │ │        ref ht       │
+    // └────────────────────────┘ └─────────────────────┘
```

Comment:
I might be mistaken here. The MVP document states that `rtt` is a subtype of `any`, which would indicate a reftype, but also states that it is a value type? Since `(ref null (rtt n ht))` seems weird, I assumed that the `any` there means the value `any`?

Reply:
rtt is a value type, but the $T in `(ref null $T)` can only be a heap type, so `(ref null (rtt ...))` is malformed.